store/sqlstore/integrity.go

```diff
@@ -0,0 +1,504 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package sqlstore
+
+import (
+	"fmt"
+
+	"github.com/mattermost/gorp"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/mattermost/mattermost-server/store"
+)
+
+type relationalCheckConfig struct {
+	parentName         string
+	parentIdAttr       string
+	childName          string
+	childIdAttr        string
+	canParentIdBeEmpty bool
+	sortRecords        bool
+}
+
+func getOrphanedRecords(dbmap *gorp.DbMap, cfg relationalCheckConfig) ([]store.OrphanedRecord, error) {
+	var records []store.OrphanedRecord
+
+	query := fmt.Sprintf(`SELECT %s AS ParentId`, cfg.parentIdAttr)
+
+	if cfg.childIdAttr != "" {
+		query += fmt.Sprintf(` , %s AS ChildId`, cfg.childIdAttr)
+	}
+
+	query += fmt.Sprintf(`
+		FROM
+			%s
+		WHERE NOT EXISTS (
+			SELECT
+  			id
```

Comment:
nit: might `SELECT TRUE` to clarify that the return value is irrelevant.

Reply:
Good point. I'll update it.